### PR TITLE
DOC update for the `run_once` sampling strategy

### PR DESCRIPTION
**Pull request by MatDag**
Update the documentation for the case where `sampling_strategy` is set to `run_once`

### CODE REVIEW

**Review by tomMoral — COMMENTED** [review]
some comments

File: doc/tutorials/add_solver.rst [hunk 1]
```diff
@@ -197,6 +197,28 @@ The following snippet shows how to use the callback strategy with a user-coded s
                 self.beta -= step * grad
         ...
 
+- ``sampling_strategy = "run_once"``
+
+In some cases, we are not interested in convergence curves over time but only in the final solution.
```
Comment (mathurinm) [inline doc/tutorials/add_solver.rst:202]:
```suggestion
In some cases, we are not interested in convergence curves over time but only in the final result.
```

File: doc/user_guide/advanced.rst [hunk 2]
```diff
@@ -148,7 +148,7 @@ implemented in the ``Solver.warm_up`` method, which is empty by default and
 called after the `set_objective` method. For solvers with
 ``sampling_strategy`` in ``{'tolerance',  'iteration'}``, simply calling the
 ``Solver.run`` with a simple enough value is usually enough. For solvers with
-``sampling_strategy`` set to ``'callback'``, it is possible to call
+``sampling_strategy`` in ``{'callback', 'run_once'}``, it is possible to call
```
Comment (tomMoral) [inline doc/user_guide/advanced.rst:151]: Not sure that this is really the right message?

Comment (MatDag) [reply doc/user_guide/advanced.rst:152]: Why wouldn't that be the case?

File: doc/user_guide/advanced.rst [hunk 3]
```diff
@@ -149,7 +149,7 @@ implemented in the ``Solver.warm_up`` method, which is empty by default and
...
```
Comment (tomMoral) [inline doc/user_guide/advanced.rst:152]: for sampling strategy `run_once`, calling the  `run_once` method makes little sense no?

Comment (MatDag) [reply doc/user_guide/advanced.rst:152]: Indeed

**Review by tomMoral — APPROVED** [review]
LGTM!

**Pull request merged**